client/my-sites/test/controller.js

```diff
@@ -0,0 +1,206 @@
+/**
+ * @jest-environment jsdom
+ */
+
+/**
+ * External dependencies
+ */
+import page from 'page';
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+
+/**
+ * Internal dependencies
+ */
+import {
+	updateRecentSitesPreferences,
+	recordNoSitesPageView,
+	recordNoVisibleSitesPageView,
+	redirectToPrimary,
+} from '../controller';
+import * as pageView from 'calypso/lib/analytics/page-view';
+import { PREFERENCES_SET } from 'calypso/state/action-types';
+
+const middlewares = [ thunk ];
+const mockStore = configureStore( middlewares );
+
+describe( 'updateRecentSitesPreferences', () => {
+	it( 'should do nothing if remote preferences are not available', () => {
+		const initialState = {
+			preferences: {
+				remoteValues: null,
+			},
+		};
+		const store = mockStore( initialState );
+
+		updateRecentSitesPreferences( { store } );
+
+		expect( store.getActions() ).toHaveLength( 0 );
+	} );
+
+	it( 'should not nothing if no site is selected', () => {
```

Comment:
Just a grammar typo..  
I'm assuming it should read, "should _**do**_ nothing if.. ", 
as opposed to, "should not nothing if... ".